src/integer.c

```diff
@@ -343,26 +345,11 @@ static Obj GMPorINTOBJ_FAKEMPZ( fake_mpz_t fake )
 */
 static Obj GMPorINTOBJ_MPZ( mpz_t v )
 {
-  Int size = v->_mp_size;
-  Obj obj;
-  if ( size == 0 )
-    obj = INTOBJ_INT(0);
-  else if ( size == 1 )
-    obj = ObjInt_UInt( v->_mp_d[0] );
-  else if ( size == -1 )
-    obj = ObjInt_UIntInv( v->_mp_d[0] );
-  else {
-    Int sign = size > 0 ? 1 : -1;
+    Int size = v->_mp_size;
     if (size < 0)
-      size = -size;
-    obj = NewBag(sign == 1 ? T_INTPOS : T_INTNEG, size * sizeof(mp_limb_t));
-    memcpy(ADDR_INT(obj), v->_mp_d, size * sizeof(mp_limb_t));
-
-    // FIXME: necessary to normalize and reduce???
-    obj = GMP_NORMALIZE( obj );
-    obj = GMP_REDUCE( obj );
-  }
-  return obj;
+        return MakeObjInt((const UInt *)v->_mp_d, -size, 1);
+    else
+        return MakeObjInt((const UInt *)v->_mp_d, size, 0);
```

Comment:
Is there a specific reason to distinguish between positive and negative sizes here and not in `MakeObjInt`?

Reply:
Encoding the sign inside the size is something specific to GMP, and I wanted this API to be generic. Of course as long as all client code is using GMP, it doesn't matter... so if people prefer it, I can also drop the `isNegative` argument, and encode the sign in the size, no problem at all.

Reply:
No, I just wanted to know what the reason for this is. I simply was not sure. This is fine to me. Currently this can only be used with GMP's anyway, right?

Reply:
Well, my hope is that it could be used with other bigint implementations, but most software I interact with is using GMP... though IIRC, Bill Hart has some other bigint code...